clang/lib/CIR/Dialect/Transforms/TargetLowering/LowerFunctionInfo.h

```diff
@@ -0,0 +1,218 @@
+//==-- LowerFunctionInfo.h - Represents of function argument/return types --==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file partially mimics clang/inlcude/CodeGen/LowerFunctionInfo.h. The
+// queries are adapted to operate on the CIR dialect, however.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_CLANG_LIB_CIR_DIALECT_TRANSFORMS_TARGETLOWERING_LOWERFUNCTIONINFO_H
+#define LLVM_CLANG_LIB_CIR_DIALECT_TRANSFORMS_TARGETLOWERING_LOWERFUNCTIONINFO_H
+
+#include "MissingFeature.h"
+#include "mlir/IR/Types.h"
+#include "clang/CIR/Dialect/IR/CIRTypes.h"
+#include "llvm/Support/TrailingObjects.h"
+#include <cstdint>
+
+namespace mlir {
+namespace cir {
+
+/// Helper class to encapsulate information about how a
+/// specific ABI-independent CIR type should be passed to or returned from a
+/// function in an ABI-specific way.
```

Comment:
Same here: please somehow unify with `clang/lib/CIR/CodeGen/CIRGenFunctionInfo.h`

Reply:
Should we isolate `ABIArgInfo` in separate header too?